util.go

```diff
@@ -88,20 +102,37 @@ func (s *segmentedBuffer) Len() uint32 {
 
 // If the space to write into + current buffer size has grown to half of the window size,
 // grow up to that max size, and indicate how much additional space was reserved.
-func (s *segmentedBuffer) GrowTo(max uint32, force bool) (bool, uint32) {
+func (s *segmentedBuffer) GrowTo(force bool, now time.Time) (bool, uint32) {
```

Comment:
This logic doesn't really belong in this type. Can't we compute the target window outside this type and pass it in?